src/main_window/renderer.js

```diff
@@ -257,12 +258,16 @@ function changeTimestamp() {
 
 function getPorts() {
     SerialPort.list().then(function (ports) {
-        var returnList = "";
-        ports.forEach(function (port) {
-            returnList += "<option>" + port.path + "</option>";
-        });
-        com_ports.innerHTML = returnList;
-        com_ports.value = current_port;
+        if(ports.length !== port_count) {
```

Comment:
wrapping this whole part in an if check to see wether the port count has changed